soroban-env-common/src/env_val.rs

```diff
@@ -155,100 +161,152 @@ impl<E: Env> TryFromVal<E, u128> for RawVal {
-                    let scob = ScObject::try_from_val(&env, &ob).map_err(|_| ConversionError)?;
-                    Ok(ScVal::Object(Some(scob)))
-                },
-                Tag::Symbol => {
-                    let sym: Symbol =
-                        unsafe { <Symbol as RawValConvertible>::unchecked_from_val(val) };
-                    let str: String = sym.into_iter().collect();
-                    Ok(ScVal::Symbol(str.as_bytes().try_into()?))
-                }
-                Tag::BitSet => Ok(ScVal::Bitset(val.get_payload())),
-                Tag::Status => {
-                    let status: Status =
-                        unsafe { <Status as RawValConvertible>::unchecked_from_val(val) };
-                    Ok(status.try_into()?)
-                }
-                Tag::Reserved => Err(ConversionError),
+        match val.get_tag() {
+            Tag::False => Ok(ScVal::Bool(false)),
+            Tag::True => Ok(ScVal::Bool(true)),
+            Tag::Void => Ok(ScVal::Void),
+            Tag::Status => {
+                let status: Status =
+                    unsafe { <Status as RawValConvertible>::unchecked_from_val(val) };
+                Ok(status.try_into()?)
+            }
+            Tag::U32 => Ok(ScVal::U32(val.get_major())),
+            Tag::I32 => Ok(ScVal::I32(val.get_major() as i32)),
+            Tag::U64Small => Ok(ScVal::U64(val.get_body())),
+            Tag::I64Small => Ok(ScVal::I64(val.get_signed_body())),
+            Tag::TimepointSmall => Ok(ScVal::Timepoint(TimePoint(val.get_body()))),
+            Tag::DurationSmall => Ok(ScVal::Duration(Duration(val.get_body()))),
+            Tag::U128Small => Ok(ScVal::U128(Int128Parts {
+                hi: 0,
+                lo: val.get_body(),
+            })),
+            Tag::I128Small => Ok(ScVal::I128(Int128Parts {
+                hi: 0,
+                lo: val.get_signed_body() as u64,
+            })),
+            Tag::U256Small => todo!(),
+            Tag::I256Small => todo!(),
+            Tag::SymbolSmall => {
+                let sym: SymbolSmall =
+                    unsafe { <SymbolSmall as RawValConvertible>::unchecked_from_val(val) };
+                let str: String = sym.into_iter().collect();
+                Ok(ScVal::Symbol(str.as_bytes().try_into()?))
             }
+
+            Tag::LedgerKeyContractExecutable => Ok(ScVal::LedgerKeyContractExecutable),
+            Tag::U64Object
+            | Tag::I64Object
+            | Tag::TimepointObject
+            | Tag::DurationObject
+            | Tag::U128Object
+            | Tag::I128Object
+            | Tag::U256Object
+            | Tag::I256Object
+            | Tag::BytesObject
+            | Tag::StringObject
+            | Tag::SymbolObject
+            | Tag::VecObject
+            | Tag::MapObject
+            | Tag::ContractExecutableObject
+            | Tag::AddressObject
+            | Tag::LedgerKeyNonceObject => unreachable!(),
```

Comment:
Why are these unreachable?

Reply:
Because they correspond to cases that `classify()` returns `Some(obj)` for and thus the branch above would have been taken (unless there's a bug).